src/cms/logic.py

```diff
@@ -0,0 +1,323 @@
+__copyright__ = "Copyright 2023 Birkbeck, University of London"
+__author__ = "Joseph Muller"
+__license__ = "AGPL v3"
+__maintainer__ = "Open Library of Humanities"
+
+import json
+import os
+import time
+from urllib.parse import urljoin, urlparse
+import requests
+from bs4 import BeautifulSoup, Comment, NavigableString, Tag
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.files.base import ContentFile
+
+from cms import models as models
+from journal import models as journal_models
+from press import models as press_models
+from repository import models as repository_models
+from utils.function_cache import cache
+from utils.logger import get_logger
+
+logger = get_logger(__name__)
+
+SITE_SEARCH_PATH = os.path.join(
+    settings.MEDIA_ROOT,
+    'press',
+    settings.SITE_SEARCH_DIR,
+)
+
+search_documents = []
+fetched_urls = set()
```

Comment:
Just to avoid having a lot of params for every function. I've gone ahead and passed them around, as well as adding type annotations to make it clearer what is being passed.